image-sync/on-demand/sync.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# only run within EV2 for now
+# RH dev envs have the regular component sync in place to make images available
+if [ -z "${EV2:-}" ]; then
+    echo "The image-sync/on-demand/sync.sh script runs only within EV2. Exiting."
+    exit 0
+fi
+
+# validate
+REQUIRED_VARS=("PULL_SECRET_KV" "PULL_SECRET" "TARGET_ACR" "SOURCE_REGISTRY" "REPOSITORY" "DIGEST")
+for VAR in "${REQUIRED_VARS[@]}"; do
+    if [ -z "${!VAR}" ]; then
+        echo "Error: Environment variable $VAR is not set."
+        exit 1
+    fi
+done
+
+# create temporary FS structure
+TMP_DIR=$(mktemp -d)
+CONTAINERS_DIR="${TMP_DIR}/containers"
+AUTH_JSON="${CONTAINERS_DIR}/auth.json"
+ORAS_CACHE="${TMP_DIR}/oras-cache"
+mkdir -p ${CONTAINERS_DIR}
+mkdir -p ${ORAS_CACHE}
+trap "rm -rf ${TMP_DIR}" EXIT
+
+# get pull secret for source registry
+echo "Fetch pull secret for source registry ${SOURCE_REGISTRY} from ${PULL_SECRET_KV} KV."
+az keyvault secret download --vault-name "${PULL_SECRET_KV}" --name "${PULL_SECRET}" -e base64 --file "${AUTH_JSON}"
+
+# ACR login to target registry
+echo "Logging into target ACR ${TARGET_ACR}."
+RESPONSE=$(az acr login --name "${TARGET_ACR}" --expose-token 2>/dev/null)
```

Comment:
Don't redirect stderr, so the poor sod who has this fail gets an error message:

```sh
if output="$( az acr login --name "${TARGET_ACR}" --expose-token 2>&1 )"; then
  RESPONSE="${output}"
else
  echo "Failed to log in to ACR ${TARGET_ACR}: ${output}"
  exit 1
fi
```

Reply:
unless you expect `az acr login` to output to stderr and stdout at the same time, in which case you can do explicit redirects